infra/lambda.tf

```diff
@@ -0,0 +1,72 @@
+data "aws_ecr_repository" "user_service" {
+  name = "user_service"
+}
+
+resource "aws_lambda_function" "user_service" {
+  depends_on = [
+    null_resource.ecr_image,
+  ]
+
+  function_name = "user_service"
+  package_type  = "Image"
+  image_uri     = "${aws_ecr_repository.user_service.repository_url}:latest"
```

Comment:
As mentioned in my task, use `public.ecr.aws/lambda/python:3.12` instead.
If we want to de-couple terraform from the build process, then we run into the chicken-and-egg problem.
To overcome that, we can try using a pre-existing docker image as a `placeholder` until we replace it with via AWS cli commands to update the lambda.